doc/nrf/protocols/matter/end_product/watchdog.rst

```diff
@@ -0,0 +1,188 @@
+.. _ug_matter_device_watchdog:
+
+Matter Watchdog
+###############
+
+A watchdog is a type of hardware timer responsible for rebooting the device if it does not receive a specific signal within a designated time window.
+"Feeding" refers to the process of sending a signal to the watchdog module to reset the timer before it expires.
+In Matter samples, you can create multiple watchdog sources and assign them to specific functions.
+Each watchdog source created must be fed to reset the timer and prevent the device from rebooting.
+
+Overview
+********
+
+The Matter Watchdog is based on Zephyr's watchdog API and allows for the creation of multiple watchdog sources.
+Each Matter Watchdog source utilizes a unique Global Watchdog channel and must be fed individually.
+You can set a time window for the Watchdog timer and enable the automatic feeding feature.
+
+The feeding feature depends on establishing an additional timer that periodically executes the created feeding callback function.
+You can create your own feeding callback function and use it to call the ``Feed()`` method in a specific context.
+This approach eliminates the need to directly call the ``Feed()`` method in your code.
+
+A time window specifies the period within which the feeding signal must be sent to each Watchdog channel to reset the timer and prevent the device from rebooting.
+If the feeding signal is sent after the time window has elapsed, the device will reboot.
+
+To enable the Matter Watchdog feature, set the :kconfig:option:`CONFIG_NCS_SAMPLE_MATTER_WATCHDOG` kconfig option to ``y``.
+The feature is enabled by default for the release build type in all Matter samples and applications.
+
+To set the timeout for the Watchdog timer, configure the :kconfig:option:`CONFIG_NCS_SAMPLE_MATTER_WATCHDOG_TIMEOUT` with a value in milliseconds.
+By default, the timeout is set to 10 seconds.
+
+Creating a Matter Watchdog source
+*********************************
+
+The Matter Watchdog feature is based on the ``Nrf::WatchdogSource`` class, which is located in the :file:`samples/matter/common/watchdog/watchdog.h` file.
+Each Matter Watchdog source constructor includes two optional arguments:
+
+* ``feedingInterval``, which specifies the duration in milliseconds for automatically calling the attached feeding callback.
+* ``callback``, which is an implementation of the ``FeedingCallback`` that is called periodically at the defined ``feedingInterval``.
+
+Feeding the Matter Watchdog
+***************************
+
+Each Matter Watchdog source must be fed separately in a specific context.
+There are two options for feeding it:
+
+* Feeding automatically
+
+  You can utilize the automatic feeding feature of the Matter Watchdog to periodically feed the Matter Watchdog source within a defined time.
```

Comment:
```suggestion
Feeding automatically
  You can utilize the automatic feeding feature of the Matter Watchdog to periodically feed the Matter Watchdog source within a defined time.
```
Since there are so many steps between the top-level bullets, I think using the term definition formatting will look nicer.